apps/studio/components/interfaces/Auth/AuthProvidersForm/AuthProvidersFormValidation.tsx

```diff
@@ -564,25 +586,24 @@ const EXTERNAL_PROVIDER_AZURE = {
       type: 'string',
     },
   },
-  validationSchema: object().shape({
-    EXTERNAL_AZURE_ENABLED: boolean().required(),
-    EXTERNAL_AZURE_CLIENT_ID: string().when('EXTERNAL_AZURE_ENABLED', {
-      is: true,
-      then: (schema) => schema.required('Application (client) ID is required'),
-      otherwise: (schema) => schema,
+  validationSchema: z.discriminatedUnion(`EXTERNAL_AZURE_ENABLED`, [
+    z.object({
+      EXTERNAL_AZURE_ENABLED: z.literal(false),
     }),
-    EXTERNAL_AZURE_SECRET: string().when('EXTERNAL_AZURE_ENABLED', {
-      is: true,
-      then: (schema) => schema.required('Secret Value is required'),
-      otherwise: (schema) => schema,
+    z.object({
+      EXTERNAL_AZURE_ENABLED: z.literal(true),
+      EXTERNAL_AZURE_CLIENT_ID: z.string({
+        required_error: 'Application (client) ID is required',
+      }),
+      EXTERNAL_AZURE_SECRET: z.string({ required_error: 'Secret Value is required' }),
+      EXTERNAL_AZURE_URL: z.string().url('Must be a valid URL').optional(),
     }),
-    EXTERNAL_AZURE_URL: string().matches(urlRegex(), 'Must be a valid URL').optional(),
-  }),
+  ]),
```

Comment:
note that the schema itself doesn't determine what fields get rendered, that is determined by the "properties" key. so that being the case, we *should* be able to use `discriminatedUnion` like this to validate the "enabled/disabled" state of this form. but if it turns out that it causes an error, we should be able to just duplicate the keys and make them optional in the disabled state (where the previously saved values would be rendered but are not required)

Reply:
highlighted this because most of the providers follow this pattern with `discriminatedUnion` for conditional fields